core/Tracker/TrackerCodeGenerator.php

```diff
@@ -93,6 +115,7 @@ public function generate($idSite, $piwikUrl, $mergeSubdomains = false, $groupPag
 
         $codeImpl = array(
             'idSite'                  => $idSite,
+            // TODO why sanitizeInputValue() and not json_encode?
             'piwikUrl'                => Common::sanitizeInputValue($piwikUrl),
```

Comment:
> What I meant was that to do it properly, you'd have to concat the protocol + url first, then do the json_encode (w/o the surrounding quotes of course).

ok that's what I had in mind. Anyway this can be done later I think, I had a look at the code and it doesn't seem so obvious (maybe I'm just tired) I don't want to over-complexify this PR. This weird encoding thing was there so nothing different should happen. And we are talking of having a quote in a URL so I think it's not very usual case ;)

Good for you?